code/fireball/fireballs.cpp

```diff
@@ -373,6 +373,90 @@ static void parse_fireball_tbl(const char *table_filename)
 				// rather than just having the default model that might or might not be used
 				fi->use_3d_warp = true;
 			}
+
+			if (optional_string("$Bobboau anim:")) {
+				stuff_boolean(&fi->bobboau_anim);
+			} else {
+				fi->bobboau_anim = false;
+			}
+
+			if (optional_string("$Warp size ratio:")) {
+				stuff_float(&fi->warp_size_ratio);
+			} else {
+				fi->warp_size_ratio = 1.0f;
+			}
+
+			if (optional_string("$Flare size ratio:")) {
+				stuff_float(&fi->flare_size_ratio);
+			} else {
+				fi->flare_size_ratio = 1.5f;
+			}
+
+			if (optional_string("$Cinematic:")) {
+				stuff_boolean(&fi->cinematic);
+
+				if (optional_string("+Warp size ratio:")) {
+					stuff_float(&fi->warp_size_ratio);
+				} else {
+					fi->warp_size_ratio = 1.6f;
+				}
+
+				if (optional_string("+Flare size ratio:")) {
+					stuff_float(&fi->flare_size_ratio);
+				} else {
+					fi->flare_size_ratio = 5.3f;
+				}
+
+				// The first two values need to be implied multiples of PI
+				// for convenience. These shouldn't need to be faster than a full
+				// rotation per second, which is already ridiculous.
+				if (optional_string("+Rotation anim:")) {
+					float rot_start_slope, rot_end_slope, rot_scale;
+
+					stuff_float(&rot_start_slope);
+					stuff_float(&rot_end_slope);
+					stuff_float(&rot_scale);
```

Comment:
These three statements can be replaced with the singular `stuff_float_list()` that's tailor made just for this kind of scenario. We've got a pretty large repertoire of parsing functions so definitely no fault of your own that you didn't find it.